source/common/filter/config_discovery_impl.h

```diff
@@ -102,12 +102,15 @@ class DynamicFilterConfigProviderImpl : public DynamicFilterConfigProviderImplBa
   }
 
   void onConfigRemoved(Config::ConfigAppliedCb applied_on_all_threads) override {
+    const absl::optional<Envoy::Http::FilterFactoryCb> default_config =
+        default_configuration_ ? absl::make_optional(factory_cb_(default_configuration_.value()))
+                               : absl::nullopt;
```

Comment:
Is this default_config need to be constructed on every onConfigRemoved?
We should be able to save the `factory_cb_(default_configuration_.value())`

Reply:
I'd prefer not to save this extra field on top of the proto just for optimization. The proto already has to be saved for `applyDefaultConfiguration()`